src/Adapters/AuthorityBasedApprovalAuthorizer.php

```diff
@@ -7,17 +7,25 @@
 use MediaWiki\Page\PageIdentity;
 use MediaWiki\Permissions\Authority;
 use ProfessionalWiki\PageApprovals\Application\ApprovalAuthorizer;
+use ProfessionalWiki\PageApprovals\Application\ApproverRepository;
+use WikiPage;
 
 class AuthorityBasedApprovalAuthorizer implements ApprovalAuthorizer {
 
 	public function __construct(
-		private Authority $authority
+		private Authority $authority,
+		private ApproverRepository $approverRepository,
+		private PageCategoriesRetriever $pageCategoriesRetriever
 	) {
 	}
 
-	public function canApprove( PageIdentity $page ): bool {
-		// TODO: logic
-		return true;
+	public function canApprove( WikiPage $page ): bool {
+		$sharedCategories = array_intersect(
+			$this->approverRepository->getApproverCategories( $this->authority->getUser()->getId() ),
+			$this->pageCategoriesRetriever->getPageCategories( $page )
+		);
+
+		return $sharedCategories !== [];
```

Comment:
I had similar logic like what was added here: https://github.com/ProfessionalWiki/PageApprovals/pull/43/files#diff-e71997fafbe96a8d4e96646051b2c13fae74c6b8af4b4cec0f1e72e302bdd0b8R32-R38

I updated my version to match that, however we should replace that code with this class in a follow-up.